packages/story-editor/src/app/story/useStoryReducer/reducers/addElements.js

```diff
@@ -76,6 +76,14 @@ export const addElements = (
     addedIds.push(...nonProducts.map(({ id }) => id));
   }
 
+  // store font props at the story level
+  draft.story.fonts = updateFonts(draft.story?.fonts, page.elements);
```

Comment:
It's a bit confusing that when an `element` is added, it has a fully defined font and then it's manually removed. Shouldn't it just be without that data from the start then? 🤔 Or is this still todo?

Reply:
I think I'm getting to a point where `draft.story.fonts` won't end up being used.  Could be wrong but the direction I'm headed in can just pull from the Font Provider based on the font family.  I don't think we'll need a state lookup / syncing across pages and all that.

I think if this is indeed a solid way to go --- we can also update the default font (I tried already but need to fix a bunch of tests that use it so held off for now).  By updating the default font --- that should setup the element without those props in the first place.

So yes I think we'll be able to not add them in the first place.

Reply:
I ended up adding a reducer for the font data vs modifying addElements to add the fonts to the state https://github.com/GoogleForCreators/web-stories-wp/pull/12445/files#diff-37271db74ac066b7cf0d328988fbd657676d6a9acdbe12277c0a8d6a17e5e1c8R77